Resources/Computing/ParallelLibraries/test/Test_ParallelLibraries.py

```diff
@@ -0,0 +1,125 @@
+""" Test ParallelLibraries
+"""
+
+import pytest
+import os
+from six.moves import reload_module
+
+from DIRAC import S_OK, gLogger
+
+
+gLogger.setLevel('DEBUG')
+
+executableContent = """
+#!/bin/bash
+
+echo "hello world"
+"""
+
+srunOutput = """
+1: line1
+1: line2
+2: line1
+1: line3
+3: line1
+2: line2
+3: line2
+2: line3
+3: line3
+"""
+
+srunExpected = """
+# On node 1
+
+ line1
+ line2
+ line3
+# On node 3
+
+ line1
+ line2
+ line3
+# On node 2
+
+ line1
+ line2
+ line3
+"""
+
+
+@pytest.mark.parametrize("parallelLibrary, parameters, expectedFile, expectedContent", [
+    ('Srun', "workingDirectory='.'", './srunExec.sh', 'srun'),
+])
+def test_generateWrapper(parallelLibrary, parameters, expectedFile, expectedContent):
+  """ Test generateWrapper()
+  """
+  # Instantiate an object from parallelLibrary class
+  parallelLibraryPath = 'DIRAC.Resources.Computing.ParallelLibraries.%s' % parallelLibrary
+  plugin = __import__(parallelLibraryPath, globals(), locals(), [parallelLibrary])  # pylint: disable=unused-variable
+  # Need to be reloaded to update the mock within the module, else, it will reuse the one when loaded the first time
+  reload_module(plugin)
+  parallelLibraryStr = 'plugin.%s(%s)' % (parallelLibrary, parameters)
+  parallelLibraryInstance = eval(parallelLibraryStr)
```

Comment:
Are these necessary for all these unit tests? like a setup? You can use `@pytest.fixture` for setup/teardown.

Reply:
Alright, I used `@pytest.fixture`, thanks!